### PR TITLE
Move expressions to types and correct links

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

**Comment by kanitw** [comment]
Please merge #3275 first to avoid conflict.

File: site/docs/transform/filter.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ Vega-Lite filter transforms must have the `filter` property.
 {:#expression}
 ## Filter Expression
 
-For a [Vega Expression](https://vega.github.io/vega/docs/expressions/) string, each datum object can be referred using bound variable `datum`. For example, setting `filter` to `"datum.b2 > 60"` would make the output data includes only items that have values in the field `b2` over 60.
+For an [Expression](types.html#expression) string, each datum object can be referred using bound variable `datum`. For example, setting `filter` to `"datum.b2 > 60"` would make the output data includes only items that have values in the field `b2` over 60.
```
Comment (kanitw) [inline site/docs/transform/filter.md]: In calculate, you still use "Vega Expression".  We should be consistent :)

File: site/docs/transform/calculate.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ title: Calculate Transform
 permalink: /docs/calculate.html
 ---
 
-The formula transform extends data objects with new fields (columns) according to a calculation formula.
+The formula transform extends data objects with new fields (columns) according to a an [expression](types.html#expression).
```
Comment (kanitw) [inline site/docs/transform/calculate.md]: a an

File: site/docs/types.md [hunk 3]
```diff
@@ -5,71 +5,65 @@ title: Parameter Types
...
-- [FieldValue](#FieldValue)
+- [Any](#any)
+- [Array](#array)
+- [Boolean](#boolean)
+- [Color](#color)
+- [Number](#number)
+- [Object](#object)
+- [String](#string)
+- [URL](#url)
+- [DateTime](#datetime)
```
Comment (kanitw) [inline site/docs/types.md]: Should we order things by alphabetical order?

File: site/docs/types.md [hunk 4]
```diff
@@ -5,71 +5,65 @@ title: Parameter Types
...
+- [Expressions](#expression)
 {: .column-list }
 
 ## Literal Values
 
-<a name="*"></a><a name="Any" href="#Any">#</a>
+<a name="*"></a><a name="any" href="#any">#</a>
```
Comment (kanitw) [inline site/docs/types.md]: Please search and make sure we don't refer to these anchor names with Titlecase from other pages.

Comment (domoritz) [reply site/docs/types.md]: I searched for `types.html`.

**Comment by domoritz** [comment]
Revised

**Comment by kanitw** [comment]
Just merged #3275 -- there are a few minor conflicts.

**Comment by domoritz** [comment]
Rebased

**Pull request merged**